### PR TITLE
Use user-friendly permission labels in access-errors log

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Now that rights are more organized in the Profile class, it is relatively easy to resolve a label from a module and right value.
Allows showing a right name for non-standard rights like "Add external" for Users.
Allows showing overridden rights like "See my ticket" instead of READ (same value).
Session is closed and the language is forced to "en_GB" to standardize new names in the log. This shouldn't be an issue though as the access errors are logged when a `check*` call fails which stops the request anyways. Before the method ends, the user's language is restored just to ensure the error displayed in the web UI is in the right language.

### CODE REVIEW

File: src/Session.php [hunk 1]
```diff
@@ -1024,10 +1026,36 @@ public static function getRightNameForError(int $right): string
             UPDATENOTE => 'UPDATENOTE',
             UNLOCK => 'UNLOCK',
         ];
-        if (array_key_exists($right, $rights)) {
-            return $rights[$right];
+        // Close session and force the default language so the logged right name is standardized
+        session_write_close();
+        $current_lang = $_SESSION['glpilanguage'];
+        $_SESSION['glpilanguage'] = 'en_GB';
```
Comment (cedric-anne) [inline src/Session.php:1032]: Is this working ?
In `NotificationTemplate` class, the call to `Session::loadLanguage($language);` seems to be required to ensure that locale files are loaded.

File: src/Session.php [hunk 2]
```diff
@@ -1004,13 +1004,15 @@ public static function checkLoginUser()
...
+     * The session will be closed by this method.
      *
-     * This only works for well-known rights.
+     * @param string $module The module
      * @param int $right The right
      * @return string The right name
      * @internal No backwards compatibility promise. Use in core only.
      */
-    public static function getRightNameForError(int $right): string
+    public static function getRightNameForError(string $module, int $right): string
```
Comment (cedric-anne) [inline src/Session.php:1015]: Changing signature in a bugfixes version may break some plugins. Either we have to create a new method (e.g. `public static function getRightLabel(string $module, int $right): string`, either we have to make the new param optional:
```suggestion
    public static function getRightNameForError(int $right, ?string $module = null): string
```

Comment (cconard96) [reply src/Session.php:1015]: Even when the PHPDoc has always said there was no BC promise and it was only meant to be used in the core?

Comment (cedric-anne) [reply src/Session.php:1015]: PHPDoc was pretty clear about this, so OK to change the signature.

**Comment by trasher** [comment]
Is it possible to add a test on that?

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**